x-pack/plugins/enterprise_search/server/lib/indices/fetch_ml_inference_pipeline_processors.test.ts

```diff
@@ -218,18 +234,29 @@ const mockGetTrainedModelStats = {
   ],
 };
 
+const mockTrainedModelsInCurrentSpace = {
+  ...mockGetTrainedModelsData,
+  trained_model_configs: [
+    ...mockGetTrainedModelsData.trained_model_configs.slice(0, 3), // Remove 4th element
+    mockGetTrainedModelsData.trained_model_configs[4],
+  ],
```

Comment:
```suggestion
  trained_model_configs: [...mockGetTrainedModelsData.trained_model_configs].splice(3, 1),
```

Reply:
`splice()` mutates the array in place, but its return value is the removed elements. So I think I'd need to do splicing on a separate line, e.g.
```
const modifiedConfigs = mockGetTrainedModelsData.trained_model_configs;
modifiedConfigs.splice(3, 1);
...
// in object declaration
trained_model_configs: modifiedConfigs
```

Reply:
Ah, good call. It's not quite as pretty, dangit